visualize.py

```diff
@@ -66,30 +68,57 @@ def plot_feature_transformer(self, title):
 
             img[ii] = weights[j]
 
+        if vmin >= 0:
+            img = np.abs(img)
+            title_template = "abs(input weights) [{BASENAME}]"
+        else:
+            title_template = "input weights [{BASENAME}]"
+
+        print(" done")
+
         # Plot image.
-        plt.matshow(np.abs(img.reshape((totaldim//totalx, totalx))),
-                    vmin=0, vmax=0.5)
-        plt.colorbar()
+        plt.figure(figsize=(16, 9))
+        plt.matshow(img.reshape((totaldim//totalx, totalx)),
+                    fignum=0, vmin=vmin, vmax=vmax, cmap='jet')
+        plt.colorbar(fraction=0.046, pad=0.04)
 
-        for i in range(numx):
-            plt.axvline(x=widthx*i-0.5, color='red')
+        line_options = {'color': 'black', 'linewidth': 0.5}
+        for i in range(1, numx):
+            plt.axvline(x=widthx*i-0.5, **line_options)
 
-        for j in range(numy):
-            plt.axhline(y=widthy*j-0.5, color='red')
+        for j in range(1, numy):
+            plt.axhline(y=widthy*j-0.5, **line_options)
 
         plt.xlim([0, totalx])
         plt.ylim([totaly, 0])
-        plt.title(title)
+        plt.xticks(ticks=widthx*np.arange(1, numx) - 0.5)
+        plt.yticks(ticks=widthy*np.arange(1, numy) - 0.5)
+        plt.axis('off')
+        plt.title(title_template.format(BASENAME=basename))
+        plt.tight_layout()
+
+        # Save figure.
+        if save_fig:
+            destname = "input-weights_{}.jpg".format(basename)
```

Comment:
can we save in png format? lossless is nice for these plots.

I think it would also be nice to have the image in the same directory as the nets, probably also for the title used, in might be useful to have more of the path, otherwise all images will be named `last.nnue` in my case, while I will use `run73/run0/last.nnue` as argument.

Reply:
Thanks for feedback, will try to address it soon.

Just a note: the resolution of the image at the moment is insufficient to losslessly represent the whole input weights matrix (which is 4096x2432). Apart from png, do we also want a "full resolution" option?

Reply:
OK, didn't realize that it would result in a >20MB image, unless it is very easy, full resolution might be a bit overkill.

Reply:
Yeah, as it currently not "lossless" anyway, I wanted to save some disk space (jpg is around 1MB, png around 2MB). So I left it at jpg...

I have addressed all your other comments with the options `--save-dir` (replaces `save-fig`), `--dont-show` and `--net-name`.